pkgs/applications/office/trilium/default.nix

```diff
@@ -58,13 +67,40 @@ in {
     '';
   
     dontStrip = true;
-  
-    meta = with stdenv.lib; {
-      inherit description;
-      homepage = https://github.com/zadam/trilium;
-      license = licenses.agpl3;
-      platforms = platforms.linux;
-      maintainers = with maintainers; [ emmanuelrosa dtzWill ];
+  };
+
+
+  trilium-server = stdenv.mkDerivation rec {
+    pname = "trilium-server";
+    inherit version;
+    inherit meta;
+
+    src = fetchurl {
+      url = "https://github.com/zadam/trilium/releases/download/v${version}/trilium-linux-x64-server-${version}.tar.xz";
+      sha256 = "04xhmc60fwvv8ip8mj112z7a9x5ahp51f1hvi20sffs0685mfaj3";
     };
+
+    nativeBuildInputs = [
+      autoPatchelfHook
+    ];
+
+    buildInputs = [
+      stdenv.cc.cc.lib
+      zlib
+      libxkbfile
+    ];
+
+    installPhase = ''
+      mkdir -p $out/bin
+      mkdir -p $out/share/trilium-server
+
+      cp -r ./* $out/share/trilium-server
+      cat > $out/bin/trilium-server <<EOF
```

Comment:
I keep thinking there must be an idiomatic way of creating the launch script so that it's not necessary to generate a script within a script.

I think the problem is that upstream doesn't provide a script to run trilium-server. Their recommended deployment method seems to be Docker, in which case they run node, just like in this script. That leads me to think a patch should be used to add a launch script, but then the script would need to be patched up to take the Nix store paths into account.

Another way is to use `pkgs.writeScript`, which would create the script in the Nix store, but not at a path of your choosing. Which means you'd essentially have three packages: trilium-desktop, trilium-server, and (for example) trilium-server-launcher. This might be overkill ;)

A simple change would be to move this into the postFixup phase. It would have the same effect, but this is the phase where typically things like wrapper scripts are created:

```
diff --git a/pkgs/applications/office/trilium/default.nix b/pkgs/applications/office/trilium/default.nix
index 2c8c2b47eab..28956d2a7c7 100644
--- a/pkgs/applications/office/trilium/default.nix
+++ b/pkgs/applications/office/trilium/default.nix
@@ -98,14 +98,15 @@ in {
       mkdir -p $out/share/trilium-server
 
       cp -r ./* $out/share/trilium-server
+    '';
+
+    postFixup = ''
       cat > $out/bin/trilium-server <<EOF
       #!${stdenv.cc.shell}
       cd $out/share/trilium-server
       exec ./node/bin/node src/www
       EOF
       chmod a+x $out/bin/trilium-server
-
-      cd $out/share/trilium-server
     '';
   };
 }
```

Reply:
Moved to `postFixup`. 
I thought about using `pkgs.writeScript` but in the end the result would be more complex I believe. In the end, the service does not need either, it could just as well run `node` directly so this is mostly about user convenience. Providing an extra derivation that generated the executable script seems counter-intuitive for that.

Reply:
Yes, that's true.

To sum it up, I like how you put this all together. It's all very clean, precise, and functional. And it's got tests!!!

Thank you, very much.

Reply:
> Thank you, very much.

Thank you for reviewing :+1: